### PR TITLE
dlt_user: use posix nanosleep

### PR DESCRIPTION
**Pull request by phongt**
For "POSIX: Replace usleep with nanosleep" as [ReleaseNotes.md](https://github.com/GENIVI/dlt-daemon/blob/master/ReleaseNotes.md#2183)

### CODE REVIEW

File: src/lib/dlt_user.c [hunk 1]
```diff
@@ -2799,7 +2799,10 @@ void dlt_user_trace_network_segmented_thread(void *unused)
         if (read < 0) {
             if (errno != EINTR) {
                 dlt_vlog(LOG_WARNING, "NWTSegmented: Error while reading queue: %s\n", strerror(errno));
-                usleep(DLT_USER_MQ_ERROR_RETRY_INTERVAL);
+                struct timespec req;
```
Comment (sherkenhoff) [inline src/lib/dlt_user.c]: Can you move the variable to the beginning of its scope, i.e. just after "if"?

File: src/lib/dlt_user.c [hunk 2]
```diff
@@ -2817,7 +2820,10 @@ void dlt_user_trace_network_segmented_thread(void *unused)
         /* DLT_NW_TRACE_RESEND custom type is used to mark a resend */
         if(data->nw_trace_type == DLT_NW_TRACE_RESEND) {
             /* Sleep 100ms, to allow other process to read FIFO */
-            usleep(100 * 1000);
+            struct timespec req;
```
Comment (sherkenhoff) [inline src/lib/dlt_user.c]: Can you move the variable to the beginning of its scope, i.e. just after "if"?

File: src/lib/dlt_user.c (hunk 1 above)
Comment (ssugiura) [inline src/lib/dlt_user.c]: Please put it before dlt_vlog(). Declaration should go to the beginning.

File: src/lib/dlt_user.c [hunk 3]
```diff
@@ -2799,7 +2799,10 @@ void dlt_user_trace_network_segmented_thread(void *unused)
         if (read < 0) {
             if (errno != EINTR) {
                 dlt_vlog(LOG_WARNING, "NWTSegmented: Error while reading queue: %s\n", strerror(errno));
-                usleep(DLT_USER_MQ_ERROR_RETRY_INTERVAL);
+                struct timespec req;
+                req.tv_sec = 0;
+                req.tv_nsec = DLT_USER_MQ_ERROR_RETRY_INTERVAL * 1000;
```
Comment (ssugiura) [inline src/lib/dlt_user.c]: As user might change the value of DLT_USER_MQ_ERROR_RETRY_INTERVAL to more than 1sec, this needs to be taken care.

File: src/lib/dlt_user.c (hunk 1 above)
Comment (phongt) [reply src/lib/dlt_user.c]: done

File: src/lib/dlt_user.c (hunk 3 above)
Comment (phongt) [reply src/lib/dlt_user.c]: Do  convert

**Pull request merged**